app/views/unattended/provisioning_templates/provision/kickstart_default.erb

```diff
@@ -232,7 +232,7 @@ key --skip
 <% else -%>
 <%= @host.diskLayout %>
 <% end -%>
-<%= snippet('kickstart_rhsm') if rhel_compatible && os_major >= 9 -%>
+<%= snippet('kickstart_rhsm') if @host.operatingsystem.name == 'RedHat' && os_major >= 9 -%>
```

Comment:
I'm for creating new variable `use_rhsm = @host.operatingsystem.name == 'RedHat' && os_major >= 9` and use it where needed.

Reply:
In theory I've got this sorted.